cmake/CMakeLists.txt

```diff
@@ -520,9 +520,12 @@ include_directories(${LAMMPS_STYLE_HEADERS_DIR})
 ###########################################
 # Actually add executable and lib to build
 ############################################
+set(LAMMPS_SUFFIX "" CACHE STRING "Suffix to append to lmp binary and liblammps")
```

Comment:
I would make the `_` (underscore) implicit. `LAMMPS_MACHINE` could be the user facing variable, while the actual suffix is `_${LAMMPS_MACHINE}` if it is set, or empty otherwise.